client/core/core.go

```diff
@@ -588,6 +588,27 @@ func (dc *dexConnection) hasActiveOrders() bool {
 	return false
 }
 
+// activeOrders returns a slice of active orders and inflight orders.
+func (dc *dexConnection) activeOrders() ([]*Order, []*InFlightOrder) {
+	dc.tradeMtx.RLock()
+	defer dc.tradeMtx.RUnlock()
+
+	var activeOrders []*Order
+	for _, trade := range dc.trades {
+		if trade.isActive() {
+			activeOrders = append(activeOrders, trade.coreOrder())
+		}
+	}
+
+	var inflightOrders []*InFlightOrder
+	for _, ord := range dc.inFlightOrders {
+		activeOrder := *ord // copy inflight orders
+		inflightOrders = append(inflightOrders, &activeOrder)
```

Comment:
Don't need to copy. The fields of an `InFlightOrder` are never modified. The `tradeMtx` protects the slice.

Reply:
Updated.